.github/workflows/build.yml

```
@@ -1,5 +1,11 @@
 name: Go Build and Test
 
+permissions:
+  contents: read
+  id-token: write
+  pull-requests: read
+  actions: write
```

```suggestion
permissions:
  contents: read
  pull-requests: read
```
These permissions are enough